### PR TITLE
Move pkg/scheduler/schedulercache -> pkg/scheduler/cache

### LINKED ISSUE #63813: Move pkg/scheduler/schedulercache -> pkg/scheduler/cache
This is a small cleanup, but the package name `k8s.io/kubernetes/pkg/scheduler/schedulercache` is redundant. Let's rename it to `k8s.io/kubernetes/pkg/scheduler/cache`.

Whoever has time for this should take a look at [gomvpkg](https://godoc.org/golang.org/x/tools/cmd/gomvpkg).

/kind cleanup
/sig scheduling
/help

### PR DESCRIPTION
**Pull request by wgliang**
**What this PR does / why we need it**:
Move pkg/scheduler/schedulercache -> pkg/scheduler/cache

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #63813

**Special notes for your reviewer**:

In order to prevent name conflicts still rename the `cache` to `schedulercache`.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by neolit123** [comment]
the rename seems appropriate :+1:

**Review by yastij — APPROVED** [review]

File: pkg/scheduler/cache/cache.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func newSchedulerCache(ttl, period time.Duration, stop <-chan struct{}) *schedul
 	}
 }
 
-// Snapshot takes a snapshot of the current schedulerCache. The method has performance impact,
```
Comment (misterikkit) [inline pkg/scheduler/cache/cache.go]: This one refers to the type and not the package. Let's leave it. (but the type's name is also redundant...)

**Review by misterikkit — COMMENTED** [review]
one small nit

**Comment by misterikkit** [comment]
Thanks for doing the work. Using the import alias is a good way to minimize the impact of the change. Individual files can be updated to stop using the alias on a case by case basis over time.

File: pkg/scheduler/cache/cache.go (hunk 1 above)
Comment (wgliang) [reply pkg/scheduler/cache/cache.go]: DONE!

**Review by bsalamat — COMMENTED** [review]
/lgtm

**Comment by wgliang** [comment]
Request your approve Bobby. @bsalamat

File: test/test_owners.csv [hunk 2]
```diff
@@ -766,7 +766,7 @@ k8s.io/kubernetes/pkg/scheduler/algorithmprovider/defaults,fgrzadkowski,0,
 k8s.io/kubernetes/pkg/scheduler/api/validation,fgrzadkowski,0,
 k8s.io/kubernetes/pkg/scheduler/core,madhusudancs,1,
 k8s.io/kubernetes/pkg/scheduler/factory,fgrzadkowski,0,
-k8s.io/kubernetes/pkg/scheduler/schedulercache,fgrzadkowski,0,
+k8s.io/kubernetes/pkg/scheduler/cache,fgrzadkowski,0,
```
Comment (misterikkit) [inline test/test_owners.csv]: I think this file is expected to be sorted alphabetically.

Comment (wgliang) [reply test/test_owners.csv]: DONE!

**Review by misterikkit — APPROVED** [review]
/lgtm

**Review by bsalamat — APPROVED** [review]

**Comment by wgliang** [comment]
/cc @misterikkit 

fix conflict and PTAL

**Review by misterikkit — APPROVED** [review]
/lgtm

**Comment by wgliang** [comment]
/cc @bsalamat @k82cn 
for approve

**Comment by k82cn** [comment]
LGTM, it's ok for scheduler part; Bobby also approved.

@kubernetes/sig-node-pr-reviews , @kubernetes/sig-cli-maintainers for node/cli part :)

**Comment by wgliang** [comment]
I have some puzzled. Why does it keep reminding me of rebase, but no conflicts are found when I merge updates.

**Comment by dchen1107** [comment]
lgtm on kubelet side.

/approve

**Comment by cblecker** [comment]
needs rebase

**Comment by wgliang** [comment]
@cblecker 
Rebase done.PTAL

**Comment by idealhack** [comment]
So we are waiting for a milestone label. /cc @cblecker

**Review by bsalamat — APPROVED** [review]
/lgtm

re-applying the label

**Pull request merged**